### ISSUE TITLE
MuJS target for mpv script

### ISSUE THREAD

**Issue by noctuid** [thheller/shadow-cljs#929]
I thought it would be interesting to try to write an [mpv](https://github.com/mpv-player/mpv) plugin in clojurescript since mpv supports scripting in javascript using [mujs](https://mujs.com/). `:target :browser` works fine when I use release command to generate a single script file. Of course it's not actually a browser though, so it doesn't work with `compile` since `window`, for example, doesn't exist. It also complains about `(0, eval)`. I looked at the other targets, but if there's one that would already work, it's not clear to me. I imagine there could be a target that would also work for applications other than mpv that use mujs to allow for scripting in javascript (not sure).

**Comment by thheller** [comment]
The `:target` abstraction in shadow-cljs is made for this. There could absolutely be a dedicated target for `mujs`. It is even user extensible. Unfortunately there are just too many JS environments and its impossible for me to keep up and maintain them all. The existing targets make specific assumptions about the environment they run in so using an ill-fitting target is expected to fail. Making generic targets is really difficult given the vastly different capabilities of the runtimes.

All the currently supported target implementations are available here:
https://github.com/thheller/shadow-cljs/tree/master/src/main/shadow/build/targets

The abstraction works by repeatedly calling the `process` function with a changing `:shadow.build/stage` keyword letting the implementation do whatever it needs to do at that stage.

`:node-script` is one of the simpler ones
https://github.com/thheller/shadow-cljs/blob/f31eef20a8ab412606d9c478068af634f8c60ae2/src/main/shadow/build/targets/node_script.clj#L67-L87

Basically it first needs to configure some stuff, deciding which sources to compile and then later assembling them in a way that works best for the given target platform.

Unfortunately this stuff isn't very well documented but maybe you can get the gist of it by looking at the existing ones. `:node-script` should be close to what you actually need, just replacing the `node` specific parts with `mujs` parts. I don't know the engine or what it is capable off.
